sale_restricted_qty/models/sale.py

```diff
@@ -0,0 +1,190 @@
+# Copyright 2019 Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.tools import float_compare
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    sale_min_qty = fields.Float(
+        string="Min Qty",
+        compute="_compute_sale_restricted_qty",
+        store=True,
+        digits="Product Unit of Measure",
+    )
+    force_sale_min_qty = fields.Boolean(
+        compute="_compute_sale_restricted_qty", readonly=True, store=True
+    )
+    is_qty_less_min_qty = fields.Boolean(
+        string="Qty < Min Qty", compute="_compute_is_qty_less_min_qty"
+    )
+
+    sale_max_qty = fields.Float(
+        string="Max Qty",
+        compute="_compute_sale_restricted_qty",
+        store=True,
+        digits="Product Unit of Measure",
+    )
+    force_sale_max_qty = fields.Boolean(
+        compute="_compute_sale_restricted_qty", readonly=True, store=True
+    )
+    is_qty_bigger_max_qty = fields.Boolean(
+        string="Qty > max Qty", compute="_compute_is_qty_bigger_max_qty"
+    )
+    sale_multiple_qty = fields.Float(
+        string="Multiple Qty",
+        compute="_compute_sale_restricted_qty",
+        store=True,
+        digits="Product Unit of Measure",
+    )
+    is_qty_not_multiple_qty = fields.Boolean(
+        string="Not Multiple Qty", compute="_compute_is_qty_not_multiple_qty"
+    )
+
+    @api.constrains(
+        "product_uom_qty", "sale_min_qty", "sale_max_qty", "sale_multiple_qty"
+    )
+    def check_constraint_restricted_qty(self):
+        msg = ""
+        invaild_min_lines = []
+        line_to_test = self.filtered(
+            lambda sl: not sl.product_id.force_sale_min_qty and sl.is_qty_less_min_qty
+        )
+        for line in line_to_test:
+            invaild_min_lines.append(
+                _('Product "%(product_name)s": Min Quantity %(sale_min_qty)s.')
+                % {
+                    "product_name": line.product_id.name,
+                    "sale_min_qty": line.sale_min_qty,
+                }
+            )
+
+        if invaild_min_lines:
+            msg += _(
+                "Check minimum order quantity for this products: * \n"
+            ) + "\n ".join(invaild_min_lines)
+            msg += _(
+                "\n* If you want sell quantity less than Min Quantity"
+                ',Check "force min quatity" on product'
```

Comment:
Quite confusing.. Force min quantity seems to actually mean ignore minium quantity? 

```suggestion
                ',Check "force min quantity" on product'
```